src/tests/Microsoft.PowerFx.Core.Tests/TestRunnerTests/TestRunnerTests.cs

```diff
@@ -98,6 +98,60 @@ public void TestList()
             Assert.Equal(2, passed);
         }
 
+        // Ensure we can compare Large numbers. 
+        [Fact]
+        public void TestLargeNumber()
+        {
+            // Result of Exp(430)
+            var longForm = "5579910311786366000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000";
+            var shortForm = "5.579910311786367E+186";
+
+            var mock = new MockErrorRunner
+            {
+                _hook = (expr, setup) => FormulaValue.New(double.Parse(expr))
+            };
+
+            var runner = new TestRunner(mock);
+            runner.Tests.Add(new TestCase
+            {
+                Input = shortForm,
+                Expected = longForm
+            });
+            runner.Tests.Add(new TestCase
+            {
+                Input = longForm,
+                Expected = shortForm
+            });
+
+            var (total, failed, passed, output) = runner.RunTests();
+            Assert.Equal(2, passed);
+        }
+
+        // Ensure small numbers that are different actually do fail 
+        [Fact]
+        public void TestNumberDiff()
+        {
+            var mock = new MockErrorRunner
+            {
+                _hook = (expr, setup) => FormulaValue.New(double.Parse(expr))
+            };
+
+            var runner = new TestRunner(mock);
+            runner.Tests.Add(new TestCase
+            {
+                Input = "1.56",
+                Expected = "1.57"
+            });
+            runner.Tests.Add(new TestCase
+            {
+                Input = "1.57",
+                Expected = "1.56"
+            });
```

Comment:
Let's add tests with very large numbers as well. 5.5e186 vs 5.6e186 (the example that you have in BaseRunner.cs) should fail (it's a significant difference). Other interesting cases: positive + negative (5.5e186 vs 5.5e-186; 5.5e186 vs. -5.5e186)